Adds an option to dm help

AutumnClove opened the pull request:
After creating your pull request, tick these boxes if they are applicable to you.

- [x] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [x] I have tested my changes on Python 3.5/3.6

----

### Description
Adds an option to send the help menu directly to the user who called it. 
Useful for spam prevention. Additionally should prevent future problems with commands in dm's if more where to be added.


### Related issues (if applicable)

BabyBoySnow committed:
Adds an option to dm help

BabyBoySnow committed:
should probably allow it to actually work

BabyBoySnow committed:
I'm retarded

BabyBoySnow committed:
log it

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1061,6 +1061,7 @@ def run(self):
             log.info("  Spotify integration: " + ['Disabled', 'Enabled'][self.config._spotify])
             log.info("  Legacy skip: " + ['Disabled', 'Enabled'][self.config.legacy_skip])
             log.info("  Leave non owners: " + ['Disabled', 'Enabled'][self.config.leavenonowners])
+            log.info("  Direct message help: " + ['Disabled', 'Enabled'][self.config.leavenonowners])
```
Should not be `self.config.leavenonowners`

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1140,7 +1141,10 @@ def _gen_embed(self):
 
-        return Response(desc, reply=True, delete_after=60)
+        if self.config.dmhelp:
+            await self.safe_send_message(message.author, desc, expire_in=60)
```
It feels like we still should give acknowledgement in the user's channel that we've DM'd them?

BabyBoySnow committed:
fix log, confirm we msged the user

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1140,7 +1141,11 @@ def _gen_embed(self):
-        return Response(desc, reply=True, delete_after=60)
+        if self.config.dmhelp:
+            await self.safe_send_message(message.author, desc, expire_in=60)
+            return Response(":mailbox_with_mail:", reply=False, expire_in=15)
```
Shouldn't this be `delete_after`?

AutumnClove committed:
Return = delete after safe send = expire

AutumnClove committed:
Review (#34)

* Localization to Taiwan (traditional Chinese)

+ /config/i18n/zh_TW.json : traditional Chinese version file.

* Fixed up editing np_msg

* now playing message for automatic entries

* wording

* Configurability

* Close #526

* Close #1090

* Update zh_TW.json

update a new added strings from 
[Option to check if owner is in a server and leave if they're not. (#1597)]

* Swedish language

not tested in bot, might need to be edited to make perfect sense in sentences

* Update sv.json

* Update ISSUE_TEMPLATE.md

* Update PULL_REQUEST_TEMPLATE.md

* Started translating i18n english file to spanish

* Translating english to spanish

* Finished translating to spanish

* Spell check

* Add end of file newline

* done it

* oh no

* Update requirements.txt

* Update README.md

* close #1726

* aiohttp.Timeout seems to be aiohttp 2 things

* Update readme

* Update license file

* tested

* add codeowners

* bump version

* dummy progress counter ported from ModuBot

* per documentation

* remove session id (and other query stuff)

* update readme image

* update ffmpeg binaries

20180307-5ab0ecf -> 20190505-e384f6f (win32)

* fix perms

* update readme badges

* Fine tune how you want the bot to send np messages

* typo

* more concise

* Slight rewording

BabyBoySnow committed:
Update ffmpeg.exe

BabyBoySnow committed:
Merge branch 'patch-1' of https://github.com/AutumnClove/MusicBot into patch-1

BabyBoySnow committed:
Allow command to work in dm's

when there's an bound channel set, and allow unbounds is on.

AutumnClove committed:
Merge branch 'review' into patch-1

BabyBoySnow committed:
only respond if called in server

BabyBoySnow committed:
better embed handling

AutumnClove committed:
Merge branch 'review' into patch-1

AutumnClove committed:
Merge pull request #41 from Just-Some-Bots/review

Review

AutumnClove committed:
Merge branch 'review' into patch-1

BabyBoySnow committed:
Revert "Merge pull request #41 from Just-Some-Bots/review"

This reverts commit ddc1ccfc02025b5d997f22a1db9a85ba2a8f7e19, reversing
changes made to 120eb2886fe0f853e8b35149d9b043e117fedb0d.

BabyBoySnow committed:
Merge remote-tracking branch 'upstream/review' into review

BabyBoySnow committed:
Update bot.py

BabyBoySnow committed:
actually up to date with review

BabyBoySnow committed:
Merge remote-tracking branch 'upstream/review' into review

AutumnClove committed:
Merge branch 'master' into pr/1856

AutumnClove committed:
Merge remote-tracking branch 'origin/review' into pr/1856

AutumnClove reviewed (commented):
Pls approve

BabyBoySnow committed:
Black

AutumnClove committed:
Merge pull request #43 from Just-Some-Bots/review

Review

BabyBoySnow committed:
.logout -> .close $ asyncio.Task.all_task -> asyncio.all_task

AutumnClove committed:
Merge branch 'review' into patch-1

AutumnClove committed:
Blacked

AutumnClove committed:
Makes #1856 even to current review

AutumnClove committed:
idk man github desktop doing weird shit

AutumnClove committed:
Revert "idk man github desktop doing weird shit"

This reverts commit f5ad301f920969e813107af3c1bff37dfa8dd34f.

AutumnClove committed:
Merge branch 'patch-1' of https://github.com/AutumnClove/MusicBot into patch-1

AutumnClove committed:
Revert "Merge branch 'patch-1' of https://github.com/AutumnClove/MusicBot into patch-1"

This reverts commit 4607f7f0a37258648f198c9403fc3b2a2690a393, reversing
changes made to 72cd6d4650a93177a753fb531579e17dfdacd503.

AutumnClove committed:
Accidently had another pr in this hoping this fixes?
